sdk/formrecognizer/ai-form-recognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,125 @@
+
+### Build model error
+
+Build model errors usually occur when trying to build a custom model. The most common scenarios when this might occur is, if you are building the model with an
+[Invalid data set](#invalid-training-data-set) or an [Invalid SAS Url](#invalid-sas-url).
+
+#### Invalid training data set
+
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set, [here](https://aka.ms/customModelV3).
+
+Example error below:
+
+```bash
+RestError: Invalid argument.
+  {
+    "name": "RestError",
+    "code": "InvalidArgument",
+    "statusCode": 400,
+    .
+    .
+    .
+    "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidParameter",
+         "message": "The parameter ContainerUrl is invalid: Url is null or whitespace"
+       }
+     }
+    }
+  }
+```
+
+#### Invalid SAS URL
+
+This error points to missing permissions on the blob storage SAS URL for the Form Recognizer service to access the training dataset resource. For more information about SAS tokens for Form Recognizer, see [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/create-sas-tokens).
+
+Example error:
+
+```bash
+ RestError: Invalid argument.
+  {
+   "name": "RestError",
+   "code": "InvalidArgument",
+   "statusCode": 400,
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidSasToken",
+         "message": "The shared access signature (SAS) is invalid: SAS token expired on 10/04/2022 03:19:05 +00:00"
+       }
+     }
+   },
+   "message": "Invalid argument."
+ }
+```
+
+### Generic Error
+
+Seeing a "Generic error" returned from the SDK is most often caused by heavy load on the service. For troubleshooting issues related to service limits, see related information [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/service-limits?tabs=v30).
+
+Example error:
+
+```bash
+(3014) Generic error during training.
+Invalid model created with ID=<model ID>
+```
+
+### Unexpected time to build a custom model
+
+It is common to have a longer completion time than what is expected when building a custom model using the `neural` build mode with `DocumentModelAdministrationClient#beginBuildDocumentModel()`. Depending on the service load you can usually expect it to take ~10min.
+
+For simpler use-cases, you can use `template` build mode which uses a different model building algorithm that takes less time (typically a few seconds to build). See more about `template` custom models [here](https://aka.ms/custom-template-models). To see more information about `neural` custom models (these models use deep learning to train and build), see documentation [here](https://aka.ms/custom-neural-models).
```

Comment:
```suggestion
For simpler use-cases, you can use `template` build mode which uses a different model building algorithm that takes less time. See more about `template` custom models [here](https://aka.ms/custom-template-models). To see more information about `neural` custom models (these models use deep learning to train and build), see documentation [here](https://aka.ms/custom-neural-models).
```

Let's not make insinuations about the proper training time. This should be highly variable depending on the nature of the documents and the size of the training set. Our test models train in a few seconds, but that doesn't mean that a model will _in general_.

Customers can get a sense of whether or not the poller is operating locally by adding a state handler to the poller.

Reply:
Copied this from Python/Java guides.

cc @catalinaperalta @samvaity